config/registry/default_schema.yaml

```diff
@@ -0,0 +1,693 @@
+        primary_key: true
+        nullable: false
+      - !Column &visit
+        name: visit
+        type: int
+        primary_key: true
+        nullable: false
+      - !Column
+        name: physical_filter
+        type: string
+        nullabe: false
+      - !Column
+        name: datetime_begin
+        type: datetime
+        doc: >
+          This should be the same as the datetime_begin field of the first Exposure
+          associated with this Visit.
+      - !Column
+        name: exposure_time
+        type: float
+        doc: >
+          This should cover the full time from the start of the first Exposure to
+          the end of the last one.
+      - !Column
+        name: region
+        type: blob
+      constraints:
+      - !ForeignKeyConstraint
+        src: camera
+        tgt: Camera.camera
+      - !ForeignKeyConstraint
+        src:
+        - camera
+        - physical_filter
+        tgt:
+        - PhysicalFilter.camera
+        - name
+  
+    - !DataUnit
+      name: Exposure
+      dependencies:
+      - Camera
+      doc: >
+        An observation associated with a particular camera.  All direct observations
+        are identified with an Exposure, but derived datasets that may be based on
+        more than one Exposure (e.g. multiple snaps) are typically identified with
+        Visits instead, even for cameras that don't have multiple Exposures per
+        Visit.  The SQL representation is a table with Exposure metadata and fields
+        indicating the relationships to Visits.
+      columns:
+      - !Column
+        name: camera
+        type: string
+        primary_key: true
+        nullable: false
+      - !Column &exposure
+        name: exposure
+        type: int
+        primary_key: true
+        nullable: false
+      - !Column
+        name: visit
+        type: int
+        primary_key: true
+        doc: >
+          ID of the Visit this Exposure is associated with.  Science observations
+          should essentially always be associated with a visit, but calibration
+          observations may not be.
+      - !Column
+        name: snap
+        type: int
+        doc: >
+          If visit is not null, the index of this Exposure in the Visit, starting
+          from zero.
+      - !Column
+        name: datetime_begin
+        type: datetime
+      - !Column
+        name: exposure_time
+        type: float
+      constraints:
+      - !ForeignKeyConstraint
+        src: camera
+        tgt: Camera.camera
+      - !ForeignKeyConstraint
+        src:
+        - camera
+        - visit
+        tgt:
+        - Visit.camera
+        - Visit.visit
+  
+    - !DataUnit
+      name: ExposureRange
+      dependencies:
+      - Camera
+      doc: >
+        An inclusive range of Exposures that may be open in either direction.  There
+        is no SQL representation; its fields (duplicated below) are all part of its
+        compound primary key and are hence just present in Dataset.
```

Comment:
We should have a machine-readable way to represent the fact that the columns here are just fields that must be present `Dataset`, not something that requires an new Table to be created.

Reply:
My thought on that was that if all columns are values then no table is needed and all that exist are those columns in the `Datasets` table (e.g. the respective `DataUnit` table is skipped). Otherwise a separate table is created for this `DataUnit`.

Reply:
By "values", you mean "appears in the `Dataset` table"?  I don't see those indicated some other way (though I think the condition you want might actually be "if all columns are primary keys").

In any case, even that condition isn't enough, because it can't distinguish between "have a definition-only table" and "no table".